src/hash/rfc6234/sha.h

```diff
@@ -0,0 +1,355 @@
+/**************************** sha.h ****************************/
+/***************** See RFC 6234 for details. *******************/
+/*
+   Copyright (c) 2011 IETF Trust and the persons identified as
+   authors of the code.  All rights reserved.
+
+   Redistribution and use in source and binary forms, with or
+   without modification, are permitted provided that the following
+   conditions are met:
+
+   - Redistributions of source code must retain the above
+     copyright notice, this list of conditions and
+     the following disclaimer.
+
+   - Redistributions in binary form must reproduce the above
+     copyright notice, this list of conditions and the following
+     disclaimer in the documentation and/or other materials provided
+     with the distribution.
```

Comment:
The libgit2 linking exception does not require this so using this would require a license change?

https://github.com/amosnier/sha-2 have a more permissive license, but I haven't checked to see if the code is usable.

Reply:
I don't think so.  I think that our code is still GPL + Linking Exception but this component is license compatible with us. There's a lot of prior art here that we've included already.  (See the `COPYING` file.)

We do need to include this license in the `COPYING` file so that end-users can ~~bury it~~ include it in their documentation.